AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JavaScript/Admin/Employee_details.js

```diff
@@ -0,0 +1,163 @@
+    const ADDRESS = INP_EMPLOYEE_ADDRESS.value;
+    const QUALIFICATION = INP_EMPLOYEE_QUALIFICATION.value;
+    const POSITION = INP_EMPLOYEE_POSITION.value;
+    const SALARY = INP_EMPLOYEE_SALARY.value;
+    const GENDER = INP_EMPLOYEE_GENDER_MALE.checked ? 'Male' : INP_EMPLOYEE_GENDER_FEMALE.checked ? 'Female' : '';
+    //const PHOTO_URL = INP_EMPLOYEE_PHOTO.value;
+
+    if(!NAME || !AGE || !ADDRESS || !QUALIFICATION || !POSITION || !SALARY || !GENDER)
+    {
+        alert(AUTHENTICATION_MESSAGES.EmptyField);
+        return;
+    }
+
+
+    if(isNaN(AGE) || isNaN(SALARY))
+    {
+        alert(AUTHENTICATION_MESSAGES.NeumericValues);
+        return;
+    }
+
+    const employeeDetails = {
+        "Employee Name": NAME,
+        "Employee Age": AGE,
+        "Employee Address": ADDRESS,
+        "Employee Qualification": QUALIFICATION,
+        "Employee Position": POSITION,
+        "Employee Salary": SALARY,
+        "Gender": GENDER,
+        //"Employee Photo": PHOTO_URL
+    };
+
+    let details = JSON.parse(localStorage.getItem("Details")) || [];
+    details.push(employeeDetails);
+    localStorage.setItem("Details", JSON.stringify(details));
+
+    document.getElementById("employeeForm").reset();
+    EmployeeDetailMethods.Display();
+};
+
+EmployeeDetailMethods.Display = function()
+{
+    const details = JSON.parse(localStorage.getItem("Details")) || [];
+    const employeeDetailsContent = document.getElementById("employeeDetailsContent");
+    employeeDetailsContent.innerHTML = "";
+
+    if(details.length > 0)
+    {
+        details.forEach(function(detail, index)
+        {
+            const employeeDetails = document.createElement("div");
+            employeeDetails.innerHTML = `
+                <p>Name: ${detail["Employee Name"]}</p>
+                <p>Age: ${detail["Employee Age"]}</p>
+                <p>Address: ${detail["Employee Address"]}</p>
+                <p>Qualification: ${detail["Employee Qualification"]}</p>
+                <p>Position: ${detail["Employee Position"]}</p>
+                <p>Salary: ${detail["Employee Salary"]}</p>
+                <p>Gender: ${detail["Gender"]}</p>
+                
+                <button onclick="EmployeeDetailMethods.Update(${index})">Update</button>
+                <button onclick="EmployeeDetailMethods.Delete(${index})">Delete</button>
+            `;
+            employeeDetailsContent.appendChild(employeeDetails);
+        });
+    }
+};
+
+EmployeeDetailMethods.Delete = function(index)
+{
+    let details = JSON.parse(localStorage.getItem("Details")) || [];
+    details.splice(index, 1);
+    localStorage.setItem("Details", JSON.stringify(details));
+    EmployeeDetailMethods.Display();
+};
+
+EmployeeDetailMethods.Update = function(index)
+{
+    const details = JSON.parse(localStorage.getItem("Details")) || [];
+    const detailToUpdate = details[index];
+
+    INP_EMPLOYEE_NAME.value = detailToUpdate["Employee Name"];
```

Comment:
INP_EMPLOYEE_NAME.value = detailToUpdate["Employee Name"];

to

INP_EMPLOYEE_NAME.value = detailToUpdate[INP_EMPLOYEE_NAME.id];

Reply:
done